common/persistence/pinot/pinot_visibility_store_test.go

```diff
@@ -0,0 +1,1588 @@
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package pinotvisibility
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/uber/cadence/common/log/testlogger"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/uber/cadence/.gen/go/indexer"
+	"github.com/uber/cadence/common"
+	"github.com/uber/cadence/common/definition"
+	"github.com/uber/cadence/common/dynamicconfig"
+	"github.com/uber/cadence/common/log"
+	"github.com/uber/cadence/common/mocks"
+	p "github.com/uber/cadence/common/persistence"
+	pnt "github.com/uber/cadence/common/pinot"
+	"github.com/uber/cadence/common/service"
+	"github.com/uber/cadence/common/types"
+)
+
+var (
+	testIndex        = "test-index"
+	testDomain       = "test-domain"
+	testDomainID     = "bfd5c907-f899-4baf-a7b2-2ab85e623ebd"
+	testPageSize     = 10
+	testEarliestTime = int64(1547596872371000000)
+	testLatestTime   = int64(2547596872371000000)
+	testWorkflowType = "test-wf-type"
+	testWorkflowID   = "test-wid"
+	testCloseStatus  = int32(1)
+	testTableName    = "test-table-name"
+
+	testContextTimeout = 5 * time.Second
+
+	validSearchAttr = definition.GetDefaultIndexedKeys()
+)
+
+func TestRecordWorkflowExecutionStarted(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockPinotClient := pnt.NewMockGenericClient(ctrl)
+	mockProducer := &mocks.KafkaProducer{}
+	mgr := NewPinotVisibilityStore(mockPinotClient, &service.Config{
+		ValidSearchAttributes:  dynamicconfig.GetMapPropertyFn(definition.GetDefaultIndexedKeys()),
+		ESIndexMaxResultWindow: dynamicconfig.GetIntPropertyFn(3),
+	}, mockProducer, testlogger.New(t))
+	visibilityStore := mgr.(*pinotVisibilityStore)
+
+	// test non-empty request fields match
+	request := &p.InternalRecordWorkflowExecutionStartedRequest{}
+	request.WorkflowID = "wid"
+	memoBytes := []byte(`test bytes`)
+	request.Memo = p.NewDataBlob(memoBytes, common.EncodingTypeThriftRW)
```

Comment:
create all the mocks inside the test body (line 97) so they are isolated for each test case

Reply:
Moved the mocks into test body, but left request here because this is the input of the test.